### PR TITLE
Support for Files and Images, rewritten summarization functions

### PR DESCRIPTION
**Pull request by sbrugman**
- Renamed ExistingPath to File
- Renamed ImagePath to Image
- Version bump to 0.4.2
- Summaries: return series instead of dict
- Categorical: unicode counts now based on original character distribution instead of unique characters which are used as intermediate step for increased performance.
- Categorical: aggregate functions are included for string length (min, max, mean, median).
- Path: number of unique values for the path parts are returned
- Image: make Exif and Hash calculations optional. Also return width, height and area.
- File: in addition to the file_size, return creation, modification and access time (which were already returned).

### CODE REVIEW

File: src/visions/application/summaries/series/path_summary.py [hunk 1]
```diff
@@ -13,12 +13,18 @@ def path_summary(series: pd.Series) -> dict:
...
-        "suffix_counts": (series.map(lambda x: x.suffix).value_counts().to_dict()),
-        "name_counts": series.map(lambda x: x.name).value_counts().to_dict(),
-        "parent_counts": (series.map(lambda x: x.parent).value_counts().to_dict()),
-        "anchor_counts": (series.map(lambda x: x.anchor).value_counts().to_dict()),
+        "common_prefix": os.path.commonprefix(list(series)) or "No common prefix",
+        "stem_counts": series.map(lambda x: x.stem).value_counts(),
+        "suffix_counts": series.map(lambda x: x.suffix).value_counts(),
+        "name_counts": series.map(lambda x: x.name).value_counts(),
+        "parent_counts": series.map(lambda x: x.parent).value_counts(),
+        "anchor_counts": series.map(lambda x: x.anchor).value_counts(),
```
Comment (ieaves) [inline src/visions/application/summaries/series/path_summary.py:21]: I'm not particularly familiar with the summary portions of the codebase but the changes from `value_counts().to_dict()` to `value_counts()` represents a breaking API change right? Effectively a change from dict to series on the return? Is that breaking anywhere else?

File: src/visions/application/summaries/series/text_summary.py [hunk 2]
```diff
@@ -15,7 +17,82 @@ def get_character_counts(series) -> Dict[str, int]:
     Returns:
         A dict with character counts
     """
-    return dict(Counter(series.str.cat()))
+    return Counter(series.str.cat())
+
+
+def counter_to_series(counter: Counter) -> pd.Series:
+    counter_as_tuples = counter.most_common(len(counter))
```
Comment (ieaves) [inline src/visions/application/summaries/series/text_summary.py:24]: `most_common` without an argument will return the full list by default.

File: src/visions/application/summaries/series/unique_summary.py [hunk 3]
```diff
@@ -21,13 +20,15 @@ def unique_summary_complex(series: pd.Series) -> dict:
...
 
     """
-    summary = {}
-    # Until complex bug is fixed:
-    # https://github.com/pandas-dev/pandas/issues/17927
-    # https://github.com/pandas-dev/pandas/pull/27599
+
     n_unique = len(set(series[series.notna()].values))
-    summary.update({"n_unique": n_unique})
+    summary = {"n_unique": n_unique}
```
Comment (ieaves) [inline src/visions/application/summaries/series/unique_summary.py:33]: any reason not to use `nunique()`?

i.e. 

```
summary = {"n_unique": series.nunique(dropna=True)}
```

Comment (sbrugman) [reply src/visions/application/summaries/series/unique_summary.py:33]: This function is specific to complex types, where there is a bug in `nunique()`.

**Comment by ieaves** [comment]
I'm not super familiar with the summary portions of the code so this review is likely of minimal value but nothing jumped out to me!

**Pull request merged**